src/library/itunes/itunesmacosimporter.mm

```diff
@@ -0,0 +1,282 @@
+
+        queryInsertToPlaylists.bindValue(":id", playlistId);
+        queryInsertToPlaylists.bindValue(":name", playlistName);
+
+        if (!queryInsertToPlaylists.exec()) {
+            LOG_FAILED_QUERY(queryInsertToPlaylists);
+            return;
+        }
+
+        m_playlistNameByDbId.insert(playlistId, playlistName);
+        m_playlistChildsByDbId.insert(parentId, playlistId);
+
+        int i = 0;
+        for (ITLibMediaItem* item in playlist.items) {
+            if (m_cancelImport) {
+                return;
+            }
+
+            queryInsertToPlaylistTracks.bindValue(":playlist_id", playlistId);
+            queryInsertToPlaylistTracks.bindValue(
+                    ":track_id", dbIdFromPersistentId(item.persistentID));
+            queryInsertToPlaylistTracks.bindValue(":position", i);
+
+            if (!queryInsertToPlaylistTracks.exec()) {
+                LOG_FAILED_QUERY(queryInsertToPlaylistTracks);
+                return;
+            }
+
+            i++;
+        }
+    }
+
+    void importMediaItem(ITLibMediaItem* item, QSqlQuery& query) {
+        query.bindValue(":id", dbIdFromPersistentId(item.persistentID));
+        query.bindValue(":artist", qStringFrom(item.artist.name));
+        query.bindValue(":title", qStringFrom(item.title));
+        query.bindValue(":album", qStringFrom(item.album.title));
+        query.bindValue(":album_artist", qStringFrom(item.album.albumArtist));
+        query.bindValue(":genre", qStringFrom(item.genre));
+        query.bindValue(":grouping", qStringFrom(item.grouping));
+        query.bindValue(":year", static_cast<int>(item.year));
+        query.bindValue(":duration", static_cast<int>(item.totalTime / 1000));
+        query.bindValue(":location", qStringFrom([item.location path]));
+        query.bindValue(":rating", static_cast<int>(item.rating / 20));
+        query.bindValue(":comment", qStringFrom(item.comments));
+        query.bindValue(":tracknumber", static_cast<int>(item.trackNumber));
+        query.bindValue(":bpm", static_cast<int>(item.beatsPerMinute));
+        query.bindValue(":bitrate", static_cast<int>(item.bitrate));
+
+        if (!query.exec()) {
+            LOG_FAILED_QUERY(query);
+            return;
+        }
+    }
+};
+
+} // anonymous namespace
+
+ITunesMacOSImporter::ITunesMacOSImporter(LibraryFeature* parentFeature,
+        QSqlDatabase& database,
+        bool& cancelImport)
+        : m_parentFeature(parentFeature),
+          m_database(database),
+          m_cancelImport(cancelImport) {
+}
+
+ITunesImport ITunesMacOSImporter::importLibrary() {
+    @autoreleasepool {
+        ITunesImport iTunesImport;
+        iTunesImport.isMusicFolderLocatedAfterTracks = false;
+
+        NSError* error = nil;
+        ITLibrary* library = [[ITLibrary alloc] initWithAPIVersion:@"1.0"
+                                                             error:&error];
+
+        if (library) {
```

Comment:
Yeah, we probably could. I'm following Apple's example here in which they only check the library object too: https://developer.apple.com/documentation/ituneslibrary/itlibrary/1809375-initwithapiversion?language=objc

It should be safe either way since the documentation mentions that the method is guaranteed to return `nil` if the method errors, but it might be nice to have a descriptive error to present to the user.

Reply:
> It should be safe either way since the documentation mentions that the method is guaranteed to return nil if the method errors, but it might be nice to have a descriptive error to present to the user.

Yes, at least it would be nice to log the issue to console.

Reply:
I've added a `qWarning` that logs the localized description of the error in 88933edcd28bf37d2aade6666481902f7f3399e5